libevmasm/Assembly.cpp

```diff
@@ -74,6 +75,176 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::importAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solThrowIf(!_code.isArray(), AssemblyImportException, "Supplied JSON is not an array.");
+	for (Json::Value const& jsonItem: _code)
+		m_items.emplace_back(createAssemblyItemFromJSON(jsonItem));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// During the assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we just ignore exactly these `JUMPDEST`'s during import.
+		auto const nextIter = next(current);
+		if (nextIter != m_items.end() && current->type() == AssemblyItemType::Tag
+			&& nextIter->type() == AssemblyItemType::Operation && nextIter->instruction() == Instruction::JUMPDEST)
+			m_items.erase(nextIter);
+	}
+
+	// All other `JUMPDEST`'s that where not followed by a tag are treated as a fatal error.
+	for (auto const& item: m_items)
+		if (item.type() == AssemblyItemType::Operation && item.instruction() == Instruction::JUMPDEST)
+			throw langutil::Error(
+				1285_error,
+				langutil::Error::Type::FatalError,
+				"JUMPDEST instruction found that was not followed by tag."
+			);
+}
+
+AssemblyItem Assembly::createAssemblyItemFromJSON(Json::Value const& _json)
+{
+	solThrowIf(!_json.isObject(), AssemblyImportException, "Supplied JSON is not an object.");
+	static set<string> validMembers{"name", "begin", "end", "source", "value", "modifierDepth", "jumpType"};
+	for (auto const& member: _json.getMemberNames())
+		solThrowIf(validMembers.count(member) == 0, AssemblyImportException, "Unknown member '" + member + "'.");
```

Comment:
You seem to be referring to these as "attributes" below.
```suggestion
		solThrowIf(validAttributes.count(member) == 0, AssemblyImportException, "Unknown attribute '" + attribute + "'.");
```

Same in `fromJSON()` below.

Reply:
Will now use `member` as the term here.